### PR TITLE
feat: Add storing of AvailableChargeLinksData

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

Part of referenced story

This one creates the repository and integrates it in the already coded flow

## References

* #649

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CREATE TABLE [Charges].ChargeLinkTransmissionRequest (
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql:1]: Is this the best name? What does "transmission" refer to?
How about perhaps "messagehub.AvailableChargeLinksData"

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	Id uniqueidentifier not null primary key,
+    Recipient varchar(70) not null,
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql:3]: Is this the name? If so perhaps "RecipientName".

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinkTransmissionRequest/ChargeLinkTransmissionRequest.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+// limitations under the License.
+
+using System;
+using GreenEnergyHub.Charges.Domain.Charges;
+using GreenEnergyHub.Charges.Domain.MarketParticipants;
+using NodaTime;
+
+namespace GreenEnergyHub.Charges.Domain.ChargeLinkTransmissionRequest
+{
+    public class ChargeLinkTransmissionRequest
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinkTransmissionRequest/ChargeLinkTransmissionRequest.cs:22]: See sql comments regarding naming. Code comments on the props would be nice. They are not all obvious to me.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinkTransmissionRequest/IChargeLinkTransmissionRequestRepository.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+namespace GreenEnergyHub.Charges.Domain.ChargeLinkTransmissionRequest
+{
+    public interface IChargeLinkTransmissionRequestRepository
+    {
+        Task StoreAsync(
+            ChargeLinkCommandAcceptedEvent chargeLinkCommandAcceptedEvent,
+            MarketParticipant marketParticipant,
+            Guid messageHubId);
+
+        Task<List<ChargeLinkTransmissionRequest>> GetChargeLinkTransmissionRequestsAsync(IEnumerable<Guid> postOfficeIds);
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/ChargeLinkTransmissionRequest/IChargeLinkTransmissionRequestRepository.cs:30]: post office...

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql (hunk 2 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql]: Renaming to RecipientId

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Repositories/ChargeLinkTransmissionRequestRepository.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            MarketParticipant marketParticipant,
+            Guid messageHubId)
+        {
+            var chargeLinkTransmissionRequest =
+                _chargeLinkTransmissionRequestFactory.MapChargeLinkCommandAcceptedEvent(chargeLinkCommandAcceptedEvent, marketParticipant, messageHubId);
+            await _context.ChargeLinkTransmissionRequests.AddAsync(chargeLinkTransmissionRequest);
+            await _context.SaveChangesAsync();
+        }
+
+        public Task<List<ChargeLinkTransmissionRequest>> GetChargeLinkTransmissionRequestsAsync(IEnumerable<Guid> postOfficeIds)
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Repositories/ChargeLinkTransmissionRequestRepository.cs:49]: post office ...

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    RecipientRole int not null,
+    BusinessReasonCode int not null,
+    ChargeId varchar(70) not null,
+    ChargeOwner varchar(70) not null,
```
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.ApplyDBMigrationsApp/Scripts/Model/202110121245 Create ChargeLinkTransmissionRequest table.sql:7]: So you want to have two ChargeId columns? 😛

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Repositories/ChargeLinkTransmissionRequestRepository.cs (hunk 5 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Infrastructure/Repositories/ChargeLinkTransmissionRequestRepository.cs]: renamed to dataReferenceId

**Pull request merged**